internal/resolver/dns/dns_resolver.go

```diff
@@ -183,6 +189,56 @@ type dnsResolver struct {
 	// has data race with replaceNetFunc (WRITE the lookup function pointers).
 	wg                   sync.WaitGroup
 	disableServiceConfig bool
+
+	stopPolling    *grpcsync.Event
+	startedPolling bool
```

Comment:
FWIW you can still use `stopPolling != nil` as `startedPolling == true`.

But you need to do what we did before later: save the `stopPolling` pointer into a local variable so the goroutine can always reference the original `stopPolling`.

Reply:
Ok, I'm going to push back here (logically, it's mathematically unlikely for this push back to be fruitful as you have 20+ years of experience), but, I feel like this is much more readable. Reading this later and seeing d.stopPolling != nil would less sense then d.startedPolling. Or is the removed state super important? Anyways, I'm on the fence anyway I'm just going to change it.

Reply:
Changed.